lib/slugger.ts

```diff
@@ -96,8 +100,8 @@ export async function saveSlugWithRetries<D extends Document> (document: D, opti
     } catch (e) {
 
       if (e instanceof MongoError) {
-        // TODO check which index is affected
-        if (e.code === 11000) {
+
+        if (e.code === 11000 && e.message.includes(index)) {
```

Comment:
This is too simple as it'll simply do a substring match. Imagine there are two indices `slug` and `no_slug`. `slug` will now also match `no_slug`.